Plugins/GRPCProtobufGenerator/BuildPluginConfig.swift

```diff
@@ -0,0 +1,127 @@
+  /// May be specified multiple times; directories will be searched in order.
+  /// The target source directory is always appended
+  /// to the import paths.
+  var importPaths: [String]
+
+  /// The path to the `protoc` binary.
+  ///
+  /// If this is not set, Swift Package Manager will try to find the tool itself.
+  var protocPath: String?
+
+  // Codable conformance with defaults
+  enum CodingKeys: String, CodingKey {
+    case visibility
+    case server
+    case client
+    case message
+    case useAccessLevelOnImports
+    case importPaths
+    case protocPath
+  }
+
+  let defaultVisibility: GenerationConfig.Visibility = .internal
+  let defaultServer = true
+  let defaultClient = true
+  let defaultMessage = true
+  let defaultUseAccessLevelOnImports = false
+  let defaultImportPaths: [String] = []
+
+  init(from decoder: any Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+
+    self.visibility =
+      try container.decodeIfPresent(GenerationConfig.Visibility.self, forKey: .visibility)
+      ?? defaultVisibility
+    self.server = try container.decodeIfPresent(Bool.self, forKey: .server) ?? defaultServer
+    self.client = try container.decodeIfPresent(Bool.self, forKey: .client) ?? defaultClient
+    self.message = try container.decodeIfPresent(Bool.self, forKey: .message) ?? defaultMessage
+    self.useAccessLevelOnImports =
+      try container.decodeIfPresent(Bool.self, forKey: .useAccessLevelOnImports)
+      ?? defaultUseAccessLevelOnImports
+    self.importPaths =
+      try container.decodeIfPresent([String].self, forKey: .importPaths) ?? defaultImportPaths
+    self.protocPath = try container.decodeIfPresent(String.self, forKey: .protocPath)
+  }
+}
+
+extension GenerationConfig {
+  init(configurationFile: BuildPluginConfig, configurationFilePath: URL, outputPath: URL) {
+    self.visibility = configurationFile.visibility
+    self.server = configurationFile.server
+    self.client = configurationFile.client
+    self.message = configurationFile.message
+    // hard-code full-path to avoid collisions since this goes into a temporary directory anyway
+    self.fileNaming = .fullPath
+    self.useAccessLevelOnImports = configurationFile.useAccessLevelOnImports
+    self.importPaths = []
+
+    // Generate absolute paths for the imports relative to the config file in which they are specified
+    self.importPaths = configurationFile.importPaths.map { relativePath in
+      configurationFilePath.deletingLastPathComponent().relativePath + "/" + relativePath
+    }
+    self.protocPath = configurationFile.protocPath
+    self.outputPath = outputPath.relativePath
+  }
+}
+
+extension GenerationConfig.Visibility: Codable {
+  init?(rawValue: String) {
+    switch rawValue.lowercased() {
+    case "internal":
+      self = .internal
+    case "public":
+      self = .public
+    case "package":
+      self = .package
+    default:
+      return nil
+    }
+  }
+}
```

Comment:
Can this move to where `GenerationConfig.Visibility` is declared? Where possible it's best to keep conformances and extensions near their declaration.